supernode/daemon/mgr/gc/gc_task.go

```diff
@@ -0,0 +1,74 @@
+package gc
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/dragonflyoss/Dragonfly/pkg/timeutils"
+)
+
+func (gcm *Manager) gcTasks(ctx context.Context) {
+	var removedTaskCount int
+
+	// get all taskID and the corresponding accessTime
+	taskAccessMap, err := gcm.taskMgr.GetAccessTime(ctx)
+	if err != nil {
+		logrus.Errorf("failed to get accessTime map for GC: %v", err)
+		return
+	}
+
+	// range all task and determine whether it's expired
+	taskIDs := taskAccessMap.ListKeyAsStringSlice()
+	totalTaskNums := len(taskIDs)
+	for _, taskID := range taskIDs {
+		atime, err := taskAccessMap.GetAsInt64(taskID)
+		if err != nil {
+			logrus.Errorf("failed to get access time taskID(%s): %v", taskID, err)
+			continue
+		}
+		if timeutils.GetCurrentTimeMillis()-atime > gcm.taskExpireTime {
+			if !gcm.gcTask(ctx, taskID) {
+				continue
+			}
+
+			if err := taskAccessMap.Remove(taskID); err != nil {
+				logrus.Errorf("failed to remove taskID(%s): %v", taskID, err)
+				continue
+			}
+			removedTaskCount++
+		}
+	}
+
+	logrus.Infof("success to full gc task count(%d), remainder count(%d)", removedTaskCount, totalTaskNums-removedTaskCount)
+}
+
+func (gcm *Manager) gcTask(ctx context.Context, taskID string) bool {
+	// get CIDs according to the taskID
+	cids, err := gcm.dfgetTaskMgr.GetCIDsByTaskID(ctx, taskID)
+	if err != nil {
+		return false
+	}
+
+	return gcm.gcProgessCID(ctx, cids) &&
```

Comment:
I think we had better use 4 goroutine to execute the four function.
If the `gcProgessCID ` returns false, no next functions will be called. I do not think that is reasonable.